### ISSUE TITLE
Issue with Kotlin Enums

### ISSUE THREAD

**Issue by DSeeLP** [SpongePowered/Configurate#208]
```kotlin
enum class DatabaseType(val dataSourceClass: String) {
    MARIADB("org.mariadb.jdbc.MariaDbDataSource"),
    SQLITE("org.sqlite.SQLiteDataSource");
}
```
This code throws this exception:
```
[09.07 11:34:42] ERROR: java.lang.IllegalArgumentException: Configuration does not accept objects of type class de.dseelp.kotlincord.api.database.DatabaseType
[09.07 11:34:42] ERROR: 	at org.spongepowered.configurate.ScalarConfigValue.set(ScalarConfigValue.java:45)
[09.07 11:34:42] ERROR: 	at org.spongepowered.configurate.AbstractConfigurationNode.insertNewValue(AbstractConfigurationNode.java:264)
[09.07 11:34:42] ERROR: 	at org.spongepowered.configurate.AbstractConfigurationNode.raw(AbstractConfigurationNode.java:340)
[09.07 11:34:42] ERROR: 	at org.spongepowered.configurate.AbstractConfigurationNode.from(AbstractConfigurationNode.java:221)
[09.07 11:34:42] ERROR: 	at org.spongepowered.configurate.AbstractConfigurationNode.mergeFrom(AbstractConfigurationNode.java:303)
[09.07 11:34:42] ERROR: 	at org.spongepowered.configurate.AbstractConfigurationNode.mergeFrom(AbstractConfigurationNode.java:43)
[09.07 11:34:42] ERROR: 	at de.dseelp.kotlincord.api.configs.ConfigBuilder.build(Configs.kt:48)
```
This should not happen it should simply save the name like `SQLITE` and then lookup the value.
The error is thrown in this code in this line:
```kotlin
@ConfigSerializable
data class DatabaseConfig(
    val type: DatabaseType,
    val username: String,
    val password: String,
    val databaseName: String,
    val host: String = "127.0.0.1",
    val port: Int = 3306
)

class ConfigBuilder(val path: Path) {
    var defaults: CommentedConfigurationNode? = null
    fun defaults(block: CommentedConfigurationNode.() -> Unit) {
        defaults = commented(ConfigurationOptions.defaults().shouldCopyDefaults(true), block)
    }

    fun build(format: ConfigFormat<out ConfigurationLoader<*>>, copyDefaults: Boolean = true): Config {
        val loader = format.build(path)
        val rootNode = loader.load()
        if (copyDefaults) defaults?.let {
            rootNode.mergeFrom(it) // <---------------- At this line
            loader.save(rootNode)
        }
        return Config(loader, rootNode, defaults)
    }
}

@ConfigSerializable
data class DatabaseConfig(
    val type: DatabaseType,
    val username: String,
    val password: String,
    val databaseName: String,
    val host: String = "127.0.0.1",
    val port: Int = 3306
) {

    companion object {
        fun defaultDatabaseConfig(plugin: Plugin) = DatabaseConfig(SQLITE, "user", "password", plugin.name)

        fun load( // <---- This is the root call. It is invoked with the default values. `plugin.name=KotlinCord` `plugin.dataFolder=local/`
            plugin: Plugin,
            default: DatabaseConfig = defaultDatabaseConfig(plugin),
            configPath: Path = plugin.dataFolder / "database.json",
        ): DatabaseConfig {
            val cfg = config(ConfigFormat.JSON, configPath) {
                defaults {
                    get(default)
                }
            }
            return cfg.node.get<DatabaseConfig>()!!
        }
    }

    fun toDatabaseInfo(plugin: Plugin): DatabaseInfo = when (type) {
        MARIADB -> DatabaseInfo.mariadb(username, password, databaseName, host, port)
        SQLITE -> DatabaseInfo.sqlite(plugin.dataFolder / "${databaseName}.db")
    }
}
```

**Comment by TDRMinecraft** [comment]
Simple solution:

Don't use Kotlin

![Kotlin is filth](https://user-images.githubusercontent.com/38729194/128382533-68660b51-41dc-4cb4-a1fc-de37b1311603.png)

**Comment by zml2008** [comment]
@TDRMinecraft Please keep comments on the issue tracker to productive topics. Further spam will result in a loss of issue tracker access.

**Comment by zml2008** [comment]
The control flow in your code block is a touch hard to follow, but what is going on is you're not setting the "accepted types" appropriately. This is something you should be getting off your loader (there is a factory method there that given you a new node) -- otherwise the node doesn't know which types it can pass through so will assume the loader can handle any type.
